packages/gatsby/src/internal-plugins/query-runner/file-parser.js

```diff
@@ -45,24 +46,19 @@ const BABYLON_OPTIONS = {
   ],
 }
 
-const getMissingNameErrorMessage = file => report.stripIndent`
-  GraphQL definitions must be "named".
-  The query with the missing name is in ${file}.
-  To fix the query, add "query MyQueryName" to the start of your query.
-  So instead of:
-    {
-      allMarkdownRemark {
-        totalCount
-      }
+/**
+ * Add autogenerated query name if it wasn't defined by user.
+ */
+const generateQueryName = ({ def, hash, file }) => {
+  if (!def.name || !def.name.value) {
+    def.name = {
+      value: `${_.camelCase(file)}${hash}`,
+      kind: `Name`,
     }
```

Comment:
This is heart of this PR - I figured hash itself is not enough - we may have queries with same hashes in case of 2 identical queries.

Reply:
Will the hash remain consistent across Windows / macOS / Linux?

Reply:
I'm not sure if it will be consistent, but even if it's not then not sure if this would be a problem. Also `file` is absolute path right now, so this would defenitely change between platforms or even after moving your project to different directory

Reply:
👍